scripts/notebook-normalizer/qiskit_docs_notebook_normalizer/cell_output_data.py

```diff
@@ -0,0 +1,91 @@
+import base64
+from dataclasses import dataclass
+from pathlib import Path
+from subprocess import Popen, PIPE
+
+# Qiskit's QuantumCircuit.draw() results in Jupyter outputting both a `text/html` and
+# `text/plain` entry. The HTML entry has pre-applied formatting that makes sense in
+# a Jupyter notebook, but renders horribly in our app:
+# https://github.com/Qiskit/qiskit/blob/df379876ba10d6f490a96723b6dbbf723ec45d7a/qiskit/visualization/circuit/text.py#L761-L769
+#
+# So, we instead should render the `text/plain` entry rather than `text/html`.
+CIRCUIT_DRAW_HTML_PREFIX = '<pre style="word-wrap: normal;white-space: pre;background: #fff0;line-height: 1.1;font-family: &quot;Courier New&quot;,Courier,monospace">'
+
+
+@dataclass
+class SvgImage:
+    data: str
+    filepath: Path
+
+    def write(self):
+        self.filepath.write_text(self.data)
+
+
+@dataclass
+class RasterImage:
+    data: bytes
+    filepath: Path
+
+    def write(self):
+        self.filepath.write_bytes(self.data)
+
+
+Image = SvgImage | RasterImage
+
+
+def remove_circuit_drawing_html(output_data: dict) -> bool:
+    """Mutates a notebook cell if needed and returns True if any changes were made."""
+    if html := output_data.get("text/html"):
+        if html.startswith(CIRCUIT_DRAW_HTML_PREFIX):
+            del output_data["text/html"]
+            return True
+    return False
+
+
+def extract_image_output(
+    output_data: dict, filestem: Path, skip_conversion: bool
+) -> Image | None:
+    """Extract image output if one exists and mutate the cell to point to the destination image."""
+
+    def _get_image(
+        output_data: dict, filestem: Path, skip_conversion: bool
+    ) -> Image | None:
```

Comment:
Nit that it's better to either move this to an entirely separate function, or to keep this as an inner function but remove the arguments. Right now, the arguments for this nested function are "shadowing" the outer arguments. Nested functions have access to the outer scope (aka the "closure")

I'd probably move it to a dedicated function. It's easier to read and easier to test in the future.